### PR TITLE
adds ocb api e2e tests

### PR DESCRIPTION
**Pull request by cheesesashimi**
**- What I did**

Updated e2e tests for OCB API.

**- How to verify it**

Run the e2e test suite.

**- Description for the changelog**
Updates e2e tests to use OCB API

### CODE REVIEW

File: test/e2e-techpreview/onclusterbuild_test.go [hunk 1]
```diff
@@ -104,74 +104,33 @@ func runOnClusterBuildTest(t *testing.T, testOpts onClusterBuildTestOpts) string
...
-	prepareForTest(t, cs, testOpts)
+	mosc := prepareForTest(t, cs, testOpts)
 
-	optPoolIntoLayering(t, cs, testOpts.poolName)
+	t.Cleanup(createMachineOSConfig(t, cs, mosc))
 
 	t.Logf("Wait for build to start")
-	waitForPoolToReachState(t, cs, testOpts.poolName, func(mcp *mcfgv1.MachineConfigPool) bool {
-		return ctrlcommon.NewMachineOSBuildState(mcp).IsBuilding()
+	waitForMachineOSBuildToReachState(t, cs, testOpts.poolName, func(mosb *mcfgv1alpha1.MachineOSBuild) bool {
```
Comment (umohnani8) [inline test/e2e-techpreview/onclusterbuild_test.go:112]: question: I think I am missing this but where is `mosb` being generated or created?

Comment (cheesesashimi) [reply test/e2e-techpreview/onclusterbuild_test.go]: To my understanding, the MachineOSBuild object is created in response to the MachineOSConfig object.

**Comment by cheesesashimi** [comment]
/test e2e-gcp-op-techpreview

I think the reason this failed is because it leaves a MachineOSBuild object around even when all of the configs and pools related to it are deleted. I've added a post-test cleanup function to ensure that everything is removed between each test.

**Comment by cheesesashimi** [comment]
/test e2e-gcp-op-techpreview

The test cluster failed to come up for reasons that appear unrelated to this PR. Retrying.

**Comment by cheesesashimi** [comment]
These changes were incorporated into https://github.com/openshift/machine-config-operator/pull/4327, so this PR can be closed.

**Pull request closed without merging**